pkg/api/types.go

```diff
@@ -1205,6 +1221,13 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// Local represents a directory with specific topology contraints
+type LocalVolumeSource struct {
+	// The full path to the volume on the node
+	// For alpha, this path must be a file-based volume
+	Path string
```

Comment:
Had a discussion with @vishh and @saad-ali, here are our thoughts on how to support file and block based volume in a generic manner:

The basic idea is that the PVC/PV contains information about how to consume the volume (as a file or block), and the specific volume spec contains the "connection" information to the underlying storage.

Have a volumeType field in the PersistentVolume, which is set at creation time. 

```
kind: PersistentVolume
spec:
    volumeType: file or block
    local:
       # If we renamed path to sourcePath, does that help at all?
       path: /dir1
```

Then, in the PVC, the user specifies how they want to consume the volume
```
kind: PersistentVolumeClaim
spec:
    volumeType: file or block
    # This is only valid for file-based
    fsType: ext4
```

If the PVC wants file (which is today's behavior), and its bound PV is block type, then the system will create the fs.  If the bound PV is already fs type, then the system doesn't format anything.  This is different from today's model, where the fs is specified in the specific volume spec, but I think by moving the volume type into PV/PVC, that will make it simpler to support block access in the future.  This is also consistent with how CSI plans to support block access.

In the future, if PVC wants block, then it can only be bound to block-based PVs.

Please let me know if this makes sense. We can also schedule a meeting to discuss further if you like.